ext/ffi/lib.rs

```diff
@@ -260,6 +261,55 @@ enum NativeType {
   Pointer,
   Buffer,
   Function,
+  Struct(Vec<NativeType>),
+}
+
+impl NativeType {
+  fn get_size(&self) -> usize {
+    match self {
+      NativeType::Void => 0,
+      NativeType::U8 | NativeType::I8 | NativeType::Bool => 1,
+      NativeType::U16 | NativeType::I16 => 2,
+      NativeType::U32 | NativeType::I32 | NativeType::F32 => 4,
+      NativeType::U64 | NativeType::I64 | NativeType::F64 => 8,
+      NativeType::USize
+      | NativeType::ISize
+      | NativeType::Pointer
+      | NativeType::Function
+      | NativeType::Buffer => std::mem::size_of::<usize>(),
+      NativeType::Struct(_) => self.as_layout().size(),
+    }
+  }
+
+  fn as_layout(&self) -> Layout {
+    match self {
+      NativeType::Void => Layout::new::<()>(),
```

Comment:
I haven't checked how this connects to everything else yet, but is void allowed within a struct?

Void should be unreachable for the struct size calculations.

Reply:
Furthermore, test cases should be written for this.

Reply:
Trying to compute size of void will throw type error now. I hope that's fine?